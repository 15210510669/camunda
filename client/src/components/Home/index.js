/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a commercial license.
 * You may not use this file except in compliance with the commercial license.
 */

export {default as Home} from './Home';
export {default as Collection} from './Collection';
