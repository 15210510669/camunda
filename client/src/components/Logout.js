/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a commercial license.
 * You may not use this file except in compliance with the commercial license.
 */

import {useEffect} from 'react';
import {withRouter} from 'react-router-dom';
import {get} from 'request';
import {withErrorHandling} from 'HOC';
import {addNotification} from 'notifications';
import {t} from 'translation';

export function Logout({mightFail, history}) {
  useEffect(() => {
    mightFail(
      get('api/authentication/logout'),
      () => {
        addNotification({text: t('navigation.logoutSuccess')});
        history.replace('/');
      },
      () => {
        addNotification({text: t('navigation.logoutFailed'), type: 'error'});
        history.replace('/');
      }
    );
  }, [mightFail, history]);

  return null;
}

export default withRouter(withErrorHandling(Logout));
