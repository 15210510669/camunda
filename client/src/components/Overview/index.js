export {default as Overview} from './Overview';
export {default as Alerts} from './Alerts';
