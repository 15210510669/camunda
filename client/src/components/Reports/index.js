export {default as Reports} from './Reports';
export {default as Report} from './Report';
export {Filter} from './filter'
