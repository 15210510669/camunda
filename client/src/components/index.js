export {Footer} from './Footer';
export {Header, LogoutButton} from './Header';
export {Home} from './Home';
export {Login} from './Login';
export {PrivateRoute} from './PrivateRoute';
export {Dashboard} from './Dashboard';
export {Dashboards} from './Dashboards';
export {Report} from './Report';
export {Reports} from './Reports';
