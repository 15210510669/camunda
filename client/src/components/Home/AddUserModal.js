/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a commercial license.
 * You may not use this file except in compliance with the commercial license.
 */

import React, {Component} from 'react';

import {Button, LabeledInput, Modal, Form, Input} from 'components';
import {t} from 'translation';

const defaultState = {
  userName: '',
  groupName: '',
  activeInput: 'user',
  activeRole: 'viewer'
};

export default class AddUserModal extends Component {
  state = defaultState;

  onConfirm = () => {
    const {userName, groupName, activeInput, activeRole} = this.state;

    this.props.onConfirm(activeInput === 'user' ? userName : groupName, activeInput, activeRole);
    this.reset();
  };

  onClose = () => {
    this.props.onClose();
    this.reset();
  };

  reset = () => {
    this.setState(defaultState);
  };

  render() {
    const {open} = this.props;
    const {userName, groupName, activeInput, activeRole} = this.state;

    const validInput = !!(activeInput === 'user' ? userName : groupName);

    return (
      <Modal className="AddUserModal" open={open} onClose={this.onClose} onConfirm={this.onConfirm}>
        <Modal.Header>{t('home.roles.addUserTitle')}</Modal.Header>
        <Modal.Content>
          <Form>
            {t('common.selectWithoutEllipsis')}
            <Form.Group>
              <Form.Group noSpacing>
                <Input
                  type="radio"
                  onChange={() => this.setState({activeInput: 'user'})}
                  checked={activeInput === 'user'}
                />
                <LabeledInput
                  type="text"
                  label={t('common.user.id')}
                  className="userIdInput"
                  value={userName}
                  onClick={() => this.setState({activeInput: 'user'})}
                  onChange={({target: {value}}) => this.setState({userName: value})}
                  autoComplete="off"
                />
              </Form.Group>
              <Form.Group noSpacing>
                <Input
                  type="radio"
                  className="groupIdRadio"
                  onChange={() => this.setState({activeInput: 'group'})}
                  checked={activeInput === 'group'}
                />
                <LabeledInput
                  type="text"
                  label={t('common.user-group.id')}
                  className="groupIdInput"
                  value={groupName}
                  onClick={() => this.setState({activeInput: 'group'})}
                  onChange={({target: {value}}) => this.setState({groupName: value})}
                  autoComplete="off"
                />
              </Form.Group>
            </Form.Group>
            {t('home.roles.userRole')}
            <Form.Group>
              <LabeledInput
                checked={activeRole === 'viewer'}
                onChange={() => this.setState({activeRole: 'viewer'})}
                label={
                  <>
                    <h2 className="label">{t('home.roles.viewer')}</h2>
                    <p className="label">{t('home.roles.viewer-description')}</p>
                  </>
                }
                type="radio"
              />
              <LabeledInput
                checked={activeRole === 'editor'}
                onChange={() => this.setState({activeRole: 'editor'})}
                label={
                  <>
                    <h2 className="label">{t('home.roles.editor')}</h2>
                    <p className="label">{t('home.roles.editor-description')}</p>
                  </>
                }
                type="radio"
              />
              <LabeledInput
                checked={activeRole === 'manager'}
                onChange={() => this.setState({activeRole: 'manager'})}
                label={
                  <>
                    <h2 className="label">{t('home.roles.manager')}</h2>
                    <p className="label">{t('home.roles.manager-description')}</p>
                  </>
                }
                type="radio"
              />
            </Form.Group>
          </Form>
        </Modal.Content>
        <Modal.Actions>
          <Button className="cancel" onClick={this.onClose}>
            {t('common.cancel')}
          </Button>
          <Button
            variant="primary"
            color="blue"
            className="confirm"
            disabled={!validInput}
            onClick={this.onConfirm}
          >
            {t('common.add')}
          </Button>
        </Modal.Actions>
      </Modal>
    );
  }
}
