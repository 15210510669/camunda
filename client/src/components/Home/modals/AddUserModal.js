/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a commercial license.
 * You may not use this file except in compliance with the commercial license.
 */

import React from 'react';

import {Button, LabeledInput, Modal, Form, Typeahead, ErrorMessage} from 'components';
import {searchIdentities} from './service';

import {t} from 'translation';

const defaultState = {
  selectedIdentity: null,
  activeRole: 'viewer'
};

export default class AddUserModal extends React.Component {
  state = defaultState;

  onConfirm = () => {
    const {selectedIdentity, activeRole} = this.state;
    if (!this.isValid()) {
      return;
    }

    this.props.onConfirm(selectedIdentity.id, selectedIdentity.type, activeRole);
    this.reset();
  };

  onClose = () => {
    this.props.onClose();
    this.reset();
  };

  reset = () => {
    this.setState(defaultState);
  };

  alreadyExists = () => {
    const {selectedIdentity} = this.state;
    if (!selectedIdentity) {
      return false;
    }

    return this.props.existingUsers.find(
      ({identity: {id, type}}) => type === selectedIdentity.type && id === selectedIdentity.id
    );
  };

  formatTypeaheadOption = ({name, email, id, type}) => {
    const subTexts = [];
    if (name) {
      subTexts.push(email);
    }

    if (name || email) {
      subTexts.push(id);
    }

    return {
      text: name || email || id,
      tag: type === 'group' && ' (User Group)',
      subTexts
    };
  };

  isValid = () => this.state.selectedIdentity && !this.alreadyExists();

  render() {
    const {open} = this.props;
    const {selectedIdentity, activeRole} = this.state;

    const alreadyExists = this.alreadyExists();

    const validInput = this.isValid();

    return (
      <Modal className="AddUserModal" open={open} onClose={this.onClose} onConfirm={this.onConfirm}>
        <Modal.Header>{t('home.roles.addUserTitle')}</Modal.Header>
        <Modal.Content>
          <Form>
            {t('home.userTitle')}
            <Form.Group>
              <Typeahead
                placeholder={t('common.collection.addUserModal.searchPlaceholder')}
                values={searchIdentities}
                onSelect={identity => this.setState({selectedIdentity: identity})}
                formatter={this.formatTypeaheadOption}
              />
              {alreadyExists && selectedIdentity.type === 'user' && (
                <ErrorMessage>{t('home.roles.existing-user-error')}</ErrorMessage>
              )}
              {alreadyExists && selectedIdentity.type === 'group' && (
                <ErrorMessage>{t('home.roles.existing-group-error')}</ErrorMessage>
              )}
            </Form.Group>
            {t('home.roles.userRole')}
            <Form.Group>
              <LabeledInput
                checked={activeRole === 'viewer'}
                onChange={() => this.setState({activeRole: 'viewer'})}
                label={
                  <>
                    <h2 className="label">{t('home.roles.viewer')}</h2>
                    <p className="label">{t('home.roles.viewer-description')}</p>
                  </>
                }
                type="radio"
              />
              <LabeledInput
                checked={activeRole === 'editor'}
                onChange={() => this.setState({activeRole: 'editor'})}
                label={
                  <>
                    <h2 className="label">{t('home.roles.editor')}</h2>
                    <p className="label">{t('home.roles.editor-description')}</p>
                  </>
                }
                type="radio"
              />
              <LabeledInput
                checked={activeRole === 'manager'}
                onChange={() => this.setState({activeRole: 'manager'})}
                label={
                  <>
                    <h2 className="label">{t('home.roles.manager')}</h2>
                    <p className="label">{t('home.roles.manager-description')}</p>
                  </>
                }
                type="radio"
              />
            </Form.Group>
          </Form>
        </Modal.Content>
        <Modal.Actions>
          <Button className="cancel" onClick={this.onClose}>
            {t('common.cancel')}
          </Button>
          <Button
            variant="primary"
            color="blue"
            className="confirm"
            disabled={!validInput}
            onClick={this.onConfirm}
          >
            {t('common.add')}
          </Button>
        </Modal.Actions>
      </Modal>
    );
  }
}
