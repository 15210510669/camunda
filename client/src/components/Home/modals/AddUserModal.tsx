/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a proprietary license.
 * See the License.txt file for more information. You may not use this file
 * except in compliance with the proprietary license.
 */

import {useState} from 'react';
import {Button} from '@carbon/react';

import {LabeledInput, Modal, Form, UserTypeahead, User} from 'components';
import {getOptimizeProfile} from 'config';
import {t} from 'translation';

interface Role extends User {
  role: 'viewer' | 'editor' | 'manager';
}

interface AddUserModalProps {
  optimizeProfile: Awaited<ReturnType<typeof getOptimizeProfile>>;
  open: boolean;
  existingUsers: Role[];
  onConfirm: (users: Omit<User, 'id'>[]) => void;
  onClose: () => void;
}

export default function AddUserModal(props: AddUserModalProps) {
  const [users, setUsers] = useState<User[]>([]);
  const [activeRole, setActiveRole] = useState<Role['role']>('viewer');

  const onConfirm = () => {
    if (!users.length) {
      return;
    }

    props.onConfirm(users.map(({identity}) => ({role: activeRole, identity})));
    reset();
  };

  const onClose = () => {
    props.onClose();
    reset();
  };

  const reset = () => {
    setUsers([]);
    setActiveRole('viewer');
  };

  const {optimizeProfile, open, existingUsers} = props;

  return (
    <Modal className="AddUserModal" open={open} onClose={onClose} isOverflowVisible>
      <Modal.Header>
        {optimizeProfile === 'platform'
          ? t('home.roles.addUserGroupTitle')
          : t('home.roles.addUserTitle')}
      </Modal.Header>
      <Modal.Content>
        <Form>
          {optimizeProfile === 'platform' ? t('home.userGroupsTitle') : t('home.userTitle')}
          <Form.Group>
            <UserTypeahead
              users={users}
              collectionUsers={existingUsers}
              onChange={(users: User[]) => setUsers(users)}
              optionsOnly={optimizeProfile === 'cloud'}
            />
          </Form.Group>
          {t('home.roles.userRole')}
          <Form.Group>
            <LabeledInput
              checked={activeRole === 'viewer'}
              onChange={() => setActiveRole('viewer')}
              label={
                <>
                  <h2>{t('home.roles.viewer')}</h2>
                  <p>{t('home.roles.viewer-description')}</p>
                </>
              }
              type="radio"
            />
            <LabeledInput
              checked={activeRole === 'editor'}
              onChange={() => setActiveRole('editor')}
              label={
                <>
                  <h2>{t('home.roles.editor')}</h2>
                  <p>{t('home.roles.editor-description')}</p>
                </>
              }
              type="radio"
            />
            <LabeledInput
              checked={activeRole === 'manager'}
              onChange={() => setActiveRole('manager')}
              label={
                <>
                  <h2>{t('home.roles.manager')}</h2>
                  <p>{t('home.roles.manager-description')}</p>
                </>
              }
              type="radio"
            />
          </Form.Group>
        </Form>
      </Modal.Content>
      <Modal.Footer>
        <Button kind="secondary" className="cancel" onClick={onClose}>
          {t('common.cancel')}
        </Button>
        <Button className="confirm" disabled={!users.length} onClick={onConfirm}>
          {t('common.add')}
        </Button>
      </Modal.Footer>
    </Modal>
  );
}
