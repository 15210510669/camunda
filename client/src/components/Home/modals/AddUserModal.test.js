/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a commercial license.
 * You may not use this file except in compliance with the commercial license.
 */

import React from 'react';
import {shallow} from 'enzyme';

import {UserTypeahead} from 'components';

import AddUserModal from './AddUserModal';

const props = {
  open: true,
  existingUsers: [],
  onClose: jest.fn(),
  onConfirm: jest.fn(),
};

it('should match snapshot', () => {
  const node = shallow(<AddUserModal {...props} />);

  expect(node).toMatchSnapshot();
});

it('should call the onConfirm prop', () => {
  const node = shallow(<AddUserModal {...props} />);

  node.find(UserTypeahead).prop('onChange')([
    {id: 'USER:testUser', identity: {id: 'testUser', type: 'user'}},
  ]);
  node.setState({activeRole: 'editor'});

  node.find('.confirm').simulate('click');

  expect(props.onConfirm).toHaveBeenCalledWith([
    {identity: {id: 'testUser', type: 'user'}, role: 'editor'},
  ]);
});
