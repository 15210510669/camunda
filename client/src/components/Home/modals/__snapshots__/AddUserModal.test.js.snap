// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<Modal
  className="AddUserModal"
  onClose={[Function]}
  onConfirm={[Function]}
  open={true}
>
  <Component>
    Add User or User Group
  </Component>
  <Component>
    <Form>
      Users and User Groups
      <FormGroup>
        <OldTypeahead
          formatter={[Function]}
          onSelect={[Function]}
          placeholder="Search for name, email or ID ..."
          values={[Function]}
        />
      </FormGroup>
      User role
      <FormGroup>
        <ForwardRef(LabeledInput)
          checked={true}
          label={
            <React.Fragment>
              <h2>
                Viewer
              </h2>
              <p>
                Read-only access to all objects
              </p>
            </React.Fragment>
          }
          onChange={[Function]}
          type="radio"
        />
        <ForwardRef(LabeledInput)
          checked={false}
          label={
            <React.Fragment>
              <h2>
                Editor
              </h2>
              <p>
                Create, edit and delete Dashboards, Reports and Alerts
              </p>
            </React.Fragment>
          }
          onChange={[Function]}
          type="radio"
        />
        <ForwardRef(LabeledInput)
          checked={false}
          label={
            <React.Fragment>
              <h2>
                Manager
              </h2>
              <p>
                Full rights editing Collection and Users
              </p>
            </React.Fragment>
          }
          onChange={[Function]}
          type="radio"
        />
      </FormGroup>
    </Form>
  </Component>
  <Component>
    <ForwardRef(Button)
      className="cancel"
      onClick={[Function]}
    >
      Cancel
    </ForwardRef(Button)>
    <ForwardRef(Button)
      className="confirm"
      color="blue"
      disabled={true}
      onClick={[Function]}
      variant="primary"
    >
      Add
    </ForwardRef(Button)>
  </Component>
</Modal>
`;
